tests/tests/test_login.py

```diff
@@ -19,78 +19,186 @@
 
-        r = c.run('login', \
-                  '--server', 'https://mender-api-gateway', \
-                  '--skip-verify', \
-                  '--verbose', \
-                  '--username', 'user@tenant.com', \
-                  '--password', 'youcantguess')
+        r = c.run(
+            "login",
+            "--server",
+            "https://mender-api-gateway",
+            "--skip-verify",
+            "--verbose",
+            "--username",
+            "user@tenant.com",
+            "--password",
+            "youcantguess",
+        )
 
         assert r.returncode == 0, r.stderr
 
         self.__check_token_at(DEFAULT_TOKEN_PATH)
-        expect_output(r.stdout, \
-                             'creating directory',
-                             'saved token to',
-                             'login successful')
+        expect_output(
+            r.stdout, "creating directory", "saved token to", "login successful"
+        )
 
     def test_error_wrong_creds(self, single_user):
         c = cli.Cli()
 
-        r = c.run('login', \
-                  '--server', 'https://mender-api-gateway', \
-                  '--skip-verify', \
-                  '--username', 'notfound@tenant.com', \
-                  '--password', 'youcantguess')
+        r = c.run(
+            "login",
+            "--server",
+            "https://mender-api-gateway",
+            "--skip-verify",
+            "--username",
+            "notfound@tenant.com",
+            "--password",
+            "youcantguess",
+        )
 
         assert r.returncode != 0
 
-        expect_output(r.stderr, 'FAILURE: login failed with status 401')
+        expect_output(r.stderr, "FAILURE: login failed with status 401")
+
+    def _write_mender_cli_conf(self, conf):
+        """Simple utility function to write the configuration file"""
+        with open(os.getenv("HOME") + "/.mender-clirc", "w") as f:
+            try:
+                f.write(conf)
+            except Expeception as e:
+                pytest.fail("Failed to create configuration file: {}".format(e))
+
+    def test_login_from_configuration_file(self, single_user):
+        """test that the configuration file parameters are respected"""
+        # Wrong username and password
+        conf = """
+        {
+            "username": "foobar",
+            "password": "barbaz",
+            "server": "https://mender-api-gateway"
+        }
+        """
+
+        self._write_mender_cli_conf(conf)
+
+        c = cli.Cli()
+        r = c.run("login", "--skip-verify")
+
+        assert r.returncode != 0, r.stderr
+
+        # correct username and password, wrong server
+        conf = """
+        {
+            "username": "user@tenant.com",
+            "password": "youcantguess",
+            "server": "https://wrong.server.com"
+        }
+        """
+
+        self._write_mender_cli_conf(conf)
+
+        r = c.run("login", "--skip-verify")
+
+        assert r.returncode != 0, r.stderr
```

Comment:
I saw that in your last force-pushes you have removed the explicit check for the error message verification. Why?

Reply:
Well, spotted!
They we're just copy pastes from the other tests. I removed them to see if they passed on their own.
Adding proper stderr expectations now.

Reply:
> Well, spotted!

It is a new world since @olehermanse showed me the GitHub hidden feature of the "force-pushed" link. This has helped me greatly on reviews since. Only for that is worth to join CFEngine for an sprint! :)